src/peep/peep.c

```diff
@@ -9267,7 +9267,7 @@ static void get_ride_queue_end(sint16 *x, sint16 *y, sint16 *z){
 				if (!footpath_element_is_sloped(mapElement))
 					break;
 
-				if (footpath_element_get_slope_direction(mapElement) != direction)
+				if (footpath_element_get_slope_direction(mapElement) != (direction ^ 2))
 					break;
```

Comment:
No.  Line 9258 is the case where the queue path slopes up in direction, so the base height doesn't change respective of the level from the previous tile.  Line 9270 is the case where the queue path slopes down in direction (i.e. slopes up in the reverse direction ... towards the entrance), so it's base height is -2 respective to the previous tile.

I tested this out with the "problematic" ride in the save game from #3541 whose queue goes both up and down and this correctly gets the queue end.